Lib/test/test_os.py

```diff
@@ -3996,14 +3996,43 @@ def test_oserror_filename(self):
                     self.fail(f"No exception thrown by {func}")
 
 class CPUCountTests(unittest.TestCase):
+    def check_cpu_count(self, cpus):
+        if cpus is None:
+            # None is valid: os.cpu_count() cannot determine
```

Comment:
This effectively means the tests can all pass when our impementation is broken.  Our tests should assume that the return value will not be `None` on some known platforms.  At a minimum even if we don't do that, raise SkipTest when `None` is returned to indicate that no functionality testing could actually performed.

Reply:
In practice, it should never return None. Maybe it's time to enforce that and make the test fails in this case.

If we discover real cases where None is returned, we can investigate and only skip on these platforms, as you say.

Reply:
Hum, for this PR, I prefer to keep self.skipTest(). Once it will be merged, I will propose a separated PR for None.